source/contracts/reporting/Market.sol

```diff
@@ -233,24 +225,23 @@ contract Market is DelegationTarget, Typed, Initializable, Ownable, IMarket {
             return false;
         }
 
-        if (getUniverse().getForkingMarket() == this) {
+        if (getForkingMarket() == this) {
             tentativeWinningPayoutDistributionHash = getWinningPayoutDistributionHashFromFork();
         }
 
-        require(tentativeWinningPayoutDistributionHash != bytes32(0));
-
         finalPayoutDistributionHash = tentativeWinningPayoutDistributionHash;
         finalizationTime = block.timestamp;
         transferIncorrectDisputeBondsToWinningReportingToken();
+        // The validity bond is paid to the owner in any indeterminate outcome and the reporting window otherwise
+        doFeePayout(!isIndeterminate(), validityBondAttoeth);
+        // The reporter gas costs are paid to the owner if the designated report was correct and the reporting window otherwise
+        doFeePayout(finalPayoutDistributionHash == designatedReportPayoutHash, reporterCasCostsFeeAttoeth);
         reportingWindow.updateMarketPhase();
         return true;
 
-        // FIXME: when the market is finalized, we need to add `reportingTokens[finalPayoutDistributionHash].totalSupply()` to the reporting window.  This is necessary for fee collection which is a cross-market operation.
-        // TODO: figure out how to make it so fee distribution is delayed until all markets have been finalized; we can enforce it contract side and let the UI deal with the actual work
-        // FIXME: if finalPayoutDistributionHash != getIdentityDistributionId(), pay back validity bond holder
-        // FIXME: if finalPayoutDistributionHash == getIdentityDistributionId(), transfer validity bond to reportingWindow (reporter fee pot)
-        // FIXME: if designated report is wrong, transfer designated report bond to reportingWindow
-        // FIXME: if designated report is right, transfer designated report bond to market creator
+        // TODO: create a floating designated report bond and charge it for designated reports  
+        // TODO: if designated report is wrong, transfer designated report bond to reportingWindow
```

Comment:
Designated report bond is in REP, so it doesn't get transferred to the ReportingWindow.  It should be transferred to the winning reporting token I believe (verify by looking to see where dispute bonds go).